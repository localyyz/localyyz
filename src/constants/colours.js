// defined colors
let Foreground = "#FFFFFF";
let Black = "#000000";
let Grey = "rgb(200, 200, 200)";
let NearBlack = "rgb(30, 30, 30)";
let Red = "#FF0000";
let Pink = "#C73B6F";
let DarkGold = "#856f03";
let Text = NearBlack;
let Transparent = "rgba(255, 255, 255, 0)";
let BlackTransparent = "rgba(0, 0, 0, 0)";
let DarkTransparent = "rgba(0, 0, 0, 0.8)";
let WhiteTransparent = "rgba(255, 255, 255, 0.9)";
let Shadow = "rgba(0, 0, 0, 0.1)";
let LightDarkShadow = "rgba(0, 0, 0, 0.2)";
let WhiteShadow = "rgba(255, 255, 255, 0.1)";
let LightShadow = "rgba(255, 255, 255, 0.5)";
let Highlight = "rgba(255, 255, 255, 0.1)";
let LightYellow = "#FFFFE0";

let Purple = "#B721FF";
let Blue = "rgb(33, 212, 253)";

// brand colors
let PurpleRain = "#44179E";
let PurpleRainTransparent = "rgba(68, 23, 158, 0)";
let BlueSteel = "#2E62F1";
let UltraViolet = "#9013FE";
let CokeBottleGreen = "#50E3C2";
let SterlingGrey = "#D8D8D8";
let GhostGrey = "#FAFAFA";
let IntoTheGrey = "#313131";

// ken's choice of colours
let FadedGrey = "#EEEEEE";
let LightGrey = "#999999";
let Yellow = "#FFF200";
let LightBlue = "#DEEFF5";

export const Colours = {
  // backgrounds
  Background: FadedGrey,
  Foreground: Foreground,
  Accent: Pink,
  DarkGold: DarkGold,
  SubduedForeground: Grey,
  MenuBackground: Black,
  StatusBar: Black,
  Transparent: Transparent,
  BlackTransparent: BlackTransparent,
  DarkTransparent: DarkTransparent,
  WhiteTransparent: WhiteTransparent,
  Shadow: Shadow,
  LightDarkShadow: LightDarkShadow,
  WhiteShadow: WhiteShadow,
  LightShadow: LightShadow,
  Highlight: Highlight,
  Alert: LightYellow,

  // buttons
  PositiveButton: UltraViolet,
  NegativeButton: Red,
  DisabledButton: LightShadow,
  Clearance: Yellow,
  Action: LightBlue,

  // text
  Text: Text,
  AlternateText: Foreground,
  EmphasizedText: Black,
  SubduedText: LightGrey,
  OnSale: Red,

  // debug
  Facebook: "rgb(59, 89, 153)",

  // gradient
  FirstGradient: Blue,
  SecondGradient: Purple,

  // status
  Success: CokeBottleGreen,
  Fail: Red,

  // brand colors
  Primary: PurpleRain,
  Secondary: BlueSteel,
  Accented: CokeBottleGreen,

  // brand transparencies
  PrimaryTransparent: PurpleRainTransparent,

  // selectors
  Selected: NearBlack,
  Unselected: Grey
};

export default Colours;
