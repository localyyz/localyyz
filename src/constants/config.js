/*
 * @flow
 */
/*global __DEV__:true*/
export const Config = {
  DEV_USE_REMOTE_API: __DEV__ ? true : false
};

export default Config;
