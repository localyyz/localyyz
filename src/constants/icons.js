// third party
//import MaterialIcon from "react-native-vector-icons/MaterialIcons";
//import MaterialCommunityIcon from "react-native-vector-icons/MaterialCommunityIcons";
//import EntypoIcon from "react-native-vector-icons/Entypo";
//import FontAwesomeIcon from "react-native-vector-icons/FontAwesome";
import React from "react";
import IonIcon from "react-native-vector-icons/Ionicons";

export default {
  CheckMark: props => (
    <IonIcon {...props} name="ios-checkmark" />
  )
};
