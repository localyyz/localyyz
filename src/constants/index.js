/*
 * @flow
 * @providesModule localyyz/constants
 */
/*global __DEV__:true*/

import { Platform } from "react-native";

// other exports
import Sizes from "./sizes";
export { default as Sizes } from "./sizes";
export { default as Colours } from "./colours";
export { default as Styles } from "./styles";
export { default as Event } from "./postMessage";
export { default as Config } from "./config";
export { default as Icons } from "./icons";

// keys
export const GOOGLE_PLACES_KEY = "AIzaSyCyDtar76vRMUbDlv6uvTai9b_8F7_UBWE";
export const STRIPE_PUB_KEY = __DEV__
  ? "pk_test_niTKL8wfaUtS6YoRDl6iBaJr"
  : "pk_live_hWOrjVkLRkwWH866iWpqhZen";
export const STRIPE_DEF_MERCHANT_ID = "merchant.com.localyyz.toronto";

// links
export const PRIVACY_POLICY_URI
  = "https://localyyz.com/business/privacy-policy-businesses.html";
export const TERMS_AND_CONDITIONS_URI
  = "https://localyyz.com/business/terms-and-conditions-businesses.html";
export const DEV_REMOTE_API = "https://api-staging.localyyz.com";

// form labels
export const DEFAULT_ADDRESS = "required";
export const DEFAULT_ADDRESS_OPT = "optional";
export const DEFAULT_NAME = "required";
export const DEFAULT_EMAIL = "required";
export const DEFAULT_DISCOUNT_CODE = "optional";

// statuses
export const CART_STATUS_INPROGRESS = "inprogress";
export const CART_STATUS_PAYMENT_SUCCESS = "payment_success";

// sizes
export const NAVBAR_HEIGHT = 48 + Sizes.ScreenBottom;

// toggles
export const IS_DEALS_SUPPORTED = Platform.OS === "ios";
