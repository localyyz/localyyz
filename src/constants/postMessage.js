export default {
  ADD_CART: "add_to_cart"
}
