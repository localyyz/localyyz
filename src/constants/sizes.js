import { Dimensions, Platform, PixelRatio } from "react-native";
import ExtraDimensions from "react-native-extra-dimensions-android";
import { isIphoneX } from "react-native-iphone-x-helper";

// platform specific to accomodate soft buttons on android
const WIDTH
  = Platform.OS === "ios"
    ? Dimensions.get("window").width
    : ExtraDimensions.get("REAL_WINDOW_WIDTH");
const HEIGHT
  = Platform.OS === "ios"
    ? Dimensions.get("window").height
    : ExtraDimensions.get("REAL_WINDOW_HEIGHT")
      - ExtraDimensions.get("STATUS_BAR_HEIGHT")
      - ExtraDimensions.get("SOFT_MENU_BAR_HEIGHT")
      - ExtraDimensions.get("SMART_BAR_HEIGHT");

// font scaling amount
const FONT_SCALING = WIDTH > 320 ? 1 : 0.8;

export const Sizes = {
  Width: WIDTH,
  Height: HEIGHT,

  // device specific
  ScreenTop: isIphoneX() ? 44 : 30,
  ScreenBottom: isIphoneX() ? 34 : 0,

  // margins
  OuterFrame: 25,
  InnerFrame: 15,

  // text
  Oversized: 32 * FONT_SCALING,
  H1: 24 * FONT_SCALING,
  H2: 22 * FONT_SCALING,
  H3: 18 * FONT_SCALING,
  Text: 16 * FONT_SCALING,
  SmallText: 14,
  TinyText: 12,

  // weights
  Obese: "800",
  Heavy: "700",
  Bold: "600",
  Medium: "500",
  Normal: "300",
  Light: "100",

  // div
  Divider: 15,

  // decor
  RoundedBorders: 5,

  // buttons
  SquareButton: 75,
  IconButton: 20,
  Avatar: 40,
  NavLeft: Platform.OS == "ios" ? 18 : 36,

  // apple specific
  TabBarButton: 23,
  TabBarText: 10,

  // spacer
  Spacer: 1 / PixelRatio.get()
};

export default Sizes;
