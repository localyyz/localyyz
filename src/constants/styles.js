import Sizes from "./sizes";
import Colours from "./colours";

export const Styles = {
  // text
  Text: {
    fontWeight: Sizes.Normal,
    fontSize: Sizes.Text,
    color: Colours.Text,
    backgroundColor: Colours.Transparent
  },

  Medium: {
    fontWeight: Sizes.Medium
  },

  Underlined: {
    textDecorationLine: "underline"
  },

  Emphasized: {
    fontWeight: Sizes.Bold,
    color: Colours.EmphasizedText
  },

  Print: {
    fontFamily: "Chronicle Display"
  },

  Terminal: {
    fontFamily: "Menlo-Regular"
  },

  Modern: {
    fontFamily: "Montserrat-ExtraBold"
  },

  Title: {
    fontSize: Sizes.H1,
    fontWeight: Sizes.Heavy
  },

  SectionTitle: {
    fontSize: Sizes.Text,
    fontWeight: "bold",
    maxWidth: 3 * Sizes.Width / 4,
    fontFamily: "Helvetica",
    color: Colours.EmphasizedText
  },

  SectionSubtitle: {
    fontSize: Sizes.Text,
    fontFamily: "Helvetica",
    color: Colours.SubduedText
  },

  Oversized: {
    fontSize: Sizes.Oversized,
    fontWeight: Sizes.Heavy,
    lineHeight: Sizes.Oversized * 1.2
  },

  Subtitle: {
    color: Colours.SubduedText
  },

  SmallText: {
    fontSize: Sizes.SmallText
  },

  TinyText: {
    fontSize: Sizes.TinyText
  },

  PillText: {
    fontSize: Sizes.H2
  },

  TabBarText: {
    fontSize: Sizes.TabBarText,
    fontWeight: Sizes.Normal
  },

  FilterBarText: {
    fontSize: Sizes.H2
  },

  Alternate: {
    backgroundColor: Colours.Transparent,
    color: Colours.AlternateText,
    borderBottomColor: Colours.AlternateText
  },

  Subdued: {
    color: Colours.SubduedText
  },

  Center: {
    textAlign: "center"
  },

  // containers
  Card: {
    backgroundColor: Colours.Foreground,
    padding: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 2
  },

  TopSpacing: {
    marginTop: Sizes.Divider
  },

  BottomSpacing: {
    marginBottom: Sizes.Divider
  },

  BottomHalfSpacing: {
    marginBottom: Sizes.Divider / 2
  },

  EqualColumns: {
    flexDirection: "row",
    justifyContent: "space-between"
  },

  Horizontal: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "flex-start"
  },

  // combination of Text, Emphasized, and SmallText
  // requires to be manually updated here if
  // above changes
  Input: {
    flex: 1,
    color: Colours.EmphasizedText,
    backgroundColor: Colours.Transparent,
    fontSize: Sizes.Text
  },

  // fix to align indicator with icons
  IconOffset: {
    marginRight: Sizes.InnerFrame / 2.5
  },

  RoundedButton: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: Sizes.InnerFrame * 2,
    paddingVertical: Sizes.InnerFrame / 2,
    borderRadius: Sizes.InnerFrame,
    backgroundColor: Colours.PositiveButton
  },

  RoundedButtonText: {
    color: Colours.Foreground,
    fontWeight: Sizes.Bold
  },

  RoundedSubButton: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
    paddingHorizontal: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 2,
    borderRadius: Sizes.InnerFrame,
    backgroundColor: Colours.DisabledButton
  },

  Divider: {
    marginVertical: Sizes.InnerFrame * 2,
    marginRight: Sizes.Width / 2,
    left: 0,
    right: 0,
    height: 0.5,
    backgroundColor: Colours.Text
  },

  Separator: {
    paddingTop: Sizes.InnerFrame,
    paddingBottom: Sizes.OuterFrame,
    borderColor: Colours.Background,
    borderBottomWidth: 1,
    width: Sizes.Width
  },

  Overlay: {
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
    bottom: 0
  },

  Unflex: {
    flex: undefined
  }
};

export default Styles;
