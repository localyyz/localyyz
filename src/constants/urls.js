/*global __DEV__:true*/

const forcedProd = false;

const prod_api_url = "https://api.localyyz.com";
const dev_api_url = "http://localhost:5331";

export const API_URL = __DEV__ && !forcedProd ? dev_api_url : prod_api_url;
