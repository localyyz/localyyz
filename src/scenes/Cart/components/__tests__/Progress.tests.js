import React from "react";

// custom
import { renderWithLayout, findById } from "localyyz/tests";
import { Sizes } from "localyyz/constants";

// local
import ComponentWrapper from "../Progress";
const Component = ComponentWrapper.wrappedComponent || ComponentWrapper;

const DEFAULT_PROPS = {
  scenes: [
    { id: "first", label: "First scene" },
    { id: "second", label: "Second scene" },
    { id: "third", label: "Third scene", isHidden: true },
    { id: "fourth", label: "Fourth scene" }
  ],
  navigation: {},
  activeSceneId: "second"
};

describe(Component.name, () => {
  it("should render properly to snapshot", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let [instance, layout] = [c.getInstance(), c.toJSON()];

    expect(instance).toBeDefined();
    expect(layout).toMatchSnapshot();
  });

  it("should render progress items properly", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "second")).toBeDefined();
    expect(findById(layout, "second")).toMatchSnapshot();
  });

  it("should not render hidden steps", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "third")).not.toBeDefined();
  });

  it("should highlight active steps and its previous steps", () => {
    let c = renderWithLayout(<Component {...DEFAULT_PROPS} />);
    let layout = c.toJSON();

    expect(findById(layout, "first").props.style.borderBottomWidth).toBe(1);
    expect(findById(layout, "second").props.style.borderBottomWidth).toBe(1);
    expect(findById(layout, "fourth").props.style[0].borderBottomWidth).toBe(
      Sizes.Spacer
    );
  });
});
