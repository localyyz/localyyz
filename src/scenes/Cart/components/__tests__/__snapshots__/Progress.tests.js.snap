// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress should render progress items properly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "borderBottomColor": "rgb(30, 30, 30)",
      "borderBottomWidth": 1,
      "flex": 1,
      "opacity": 1,
      "paddingLeft": 7.5,
      "paddingRight": 7.5,
      "paddingVertical": 3.75,
    }
  }
  testID="second"
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0)",
          "color": "rgb(30, 30, 30)",
          "fontSize": 14,
          "fontWeight": "300",
        },
        Object {
          "color": "#000000",
          "fontWeight": "600",
        },
      ]
    }
  >
    Second scene
  </Text>
</View>
`;

exports[`Progress should render properly to snapshot 1`] = `
<View
  style={Object {}}
>
  <RCTScrollView
    data={
      Array [
        Object {
          "id": "first",
          "label": "First scene",
        },
        Object {
          "id": "second",
          "label": "Second scene",
        },
        Object {
          "id": "fourth",
          "label": "Fourth scene",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    renderItem={[Function]}
    scrollEnabled={false}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomColor": "rgb(30, 30, 30)",
              "borderBottomWidth": 1,
              "flex": 1,
              "opacity": 1,
              "paddingRight": 7.5,
              "paddingVertical": 3.75,
            }
          }
          testID="first"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "fontSize": 14,
                  "fontWeight": "300",
                },
                Object {
                  "color": "#000000",
                  "fontWeight": "600",
                },
              ]
            }
          >
            First scene
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "borderBottomColor": "rgb(30, 30, 30)",
              "borderBottomWidth": 1,
              "flex": 1,
              "opacity": 1,
              "paddingLeft": 7.5,
              "paddingRight": 7.5,
              "paddingVertical": 3.75,
            }
          }
          testID="second"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "fontSize": 14,
                  "fontWeight": "300",
                },
                Object {
                  "color": "#000000",
                  "fontWeight": "600",
                },
              ]
            }
          >
            Second scene
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            null,
          ]
        }
      >
        <View
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderBottomColor": "rgb(30, 30, 30)",
                "borderBottomWidth": 0.5,
                "flex": 1,
                "paddingVertical": 3.75,
              },
              Object {
                "paddingLeft": 7.5,
              },
              false,
              false,
            ]
          }
          testID="fourth"
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "rgba(255, 255, 255, 0)",
                  "color": "rgb(30, 30, 30)",
                  "fontSize": 14,
                  "fontWeight": "300",
                },
                false,
              ]
            }
          >
            Fourth scene
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
